process/economics/economicsData.go

```diff
@@ -270,13 +287,12 @@ func (ed *economicsData) ComputeTxFee(tx process.TransactionWithFeeHandler) *big
 			return ed.ComputeFeeForProcessing(tx, tx.GetGasLimit())
 		}
 
-		gasLimitForMoveBalance := ed.ComputeGasLimit(tx)
-		moveBalanceFee := core.SafeMul(tx.GetGasPrice(), gasLimitForMoveBalance)
+		gasLimitForMoveBalance, difference := ed.SplitTxGasInCategories(tx)
+		moveBalanceFee := core.SafeMul(ed.GasPriceForMove(tx), gasLimitForMoveBalance)
 		if tx.GetGasLimit() <= gasLimitForMoveBalance {
```

Comment:
I think I asked Robert about this if, if and how it could happen. Is it still needed?

Reply:
for strictly "==" it is needed for early exit
but for "<" I don't really see the scenario where it could happen. 
I guess the condition is more inclusive.

Reply:
there is not way to happen with <. the transaction is rejected by interceptor

Reply:
ok